railties/lib/rails/tasks/yarn.rake

```diff
@@ -9,6 +9,7 @@ namespace :yarn do
       valid_node_envs.include?(Rails.env) ? Rails.env : "production"
     end
     system({ "NODE_ENV" => node_env }, "#{Rails.root}/bin/yarn install --no-progress --frozen-lockfile")
+    exit(1) unless $?.success?
```

Comment:
```suggestion
    system({ "NODE_ENV" => node_env }, "#{Rails.root}/bin/yarn install --no-progress --frozen-lockfile") || raise
```

I think `raise` is more appropriate than `exit`. This allows to switch to the using `exception: true` option when only support Ruby >= 2.6.